src/java/org/apache/cassandra/index/sai/plan/QueryReferencingTooManyIndexesException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.plan;
+
+import org.apache.cassandra.db.RejectException;
+
+public class QueryReferencingTooManyIndexesException extends RejectException
```

Comment:
this exception extends `RejectedException`, it is different from `QueryReferencesTooManyIndexesAbortException` which extends `ReadAbortException`. Same pattern is used in tombstone case if you look closely. There are two tombstone-related exceptions, one rejecting the read and another one aborting it.